### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by DannyAziz**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the current behavior? (You can also link to an open issue here)**
A lot of arguments and functions missing from `docker`

**What is the new behavior (if this is a feature change)?**
All options and argument for high-level docker commands
Generators where applicable

**Additional info:**
Some code pulled in from: [https://github.com/withfig/autocomplete/pull/47](https://github.com/withfig/autocomplete/pull/47)

### CODE REVIEW

**Comment by DannyAziz** [comment]
[Docs](https://withfig.com/docs/autocomplete/api#arg-object) specify an `isCommand` but that seems to be the reason the linter has failed

**Comment by rolandtshen** [comment]
Just merged the updated linter into master, try merging master into your branch and re-run the linter jobs

**Comment by DannyAziz** [comment]
@rolandtshen :)

File: parser.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                token_with_type = token[len(token) - 1]
+                _token, type = token_with_type.split(' ')
+                token[len(token) - 1] = _token
+                opt_res['args'] = {'name': type}
+
+            opt_res['name'] = token
+            opt_res['description'] = description
+
+            opts.append(opt_res)
+    print(json.dumps(opts, indent=4, sort_keys=True))
```
Comment (rolandtshen) [inline parser.py:42]: This PR is great! Will add a lot of long-needed support for docker. The docker spec looks ready to merge, but what's the purpose of this parser?

Comment (DannyAziz) [reply parser.py]: Didn't mean to commit - Sorry

**Comment by DannyAziz** [comment]
@rolandtshen bump

**Comment by rolandtshen** [comment]
Thanks Danny for all the added Docker support! Merging

**Pull request merged**